src/utilities.rs

```diff
@@ -0,0 +1,60 @@
+/// Utilities for interfacing with GSL/C
+
+use std::ffi::CString;
+use std::path::Path;
+use std::fs::File;
+
+use libc::{FILE, fopen, fclose};
+
+/// A wrapper to handle I/O operations between GSL and rust
+pub struct IOStream {
+    inner: *mut FILE,
+    mode: Mode,
+}
+
+#[allow(dead_code)]
+enum Mode {
+    Write,
+    Read,
+}
+
+impl IOStream {
+    /// Open a file in write mode.
+    pub fn fwrite_handle(file: &Path) -> Result<IOStream, ::std::io::ErrorKind> {
```

Comment:
Instead of `&Path`, would be better to have:

```rust
pub fn fwrite_handle<P: AsRef<Path>>(file: &P) -> Result<IOStream, ::std::io::ErrorKind>
```